Incorrect handling of linker option

Hi,
For some reason my toolchain returns --exclude-libs option when asked for the options by c_check and f_check scripts when building Makefile.conf.
These options need to be prepended with "-Wl," (without the quotes), or the build will fail.

Cheers,
D

PS Took me a LONG to figure that out :(

Hi,I just noticed that it looks as if script didn't record the whole session.Anyway, the following is the tail of the run (where the error occurs):
gcc -O2 -DMS_ABI -DMAX_STACK_ALLOC=2048 -Wall -m64 -DF_INTERFACE_GFORT -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=4 -DASMNAME= -DASMFNAME=_ -DNAME=_ -DCNAME= -DCHAR_NAME=\"_\" -DCHAR_CNAME=\"\" -DNO_AFFINITY -I..  libopenblas.def dllinit.obj \-shared -o ../libopenblas.dll -Wl,--out-implib,../libopenblas.dll.a \-Wl,--whole-archive ../libopenblas_haswellp-r0.2.18.a -Wl,--no-whole-archive -LC:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2 -LC:/TDM-GCC-64/bin/../lib/gcc -LC:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/../../../../x86_64-w64-mingw32/lib/../lib -LC:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/../../../../lib -LC:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/../../../../x86_64-w64-mingw32/lib -LC:/TDM-GCC-64/bin/../lib/gcc/x86_64-w64-mingw32/4.9.2/../../..  -lgfortran -lmingw32 -lmoldname -lmingwex -lmsvcrt -lquadmath -lm -lmingw32 -lmoldname -lmingwex -lmsvcrt -lpthread -lmingw32 -lmoldname -lmingwex -lmsvcrt  --exclude-libs=libpthread.a  -defaultlib:advapi32 -lgfortran -defaultlib:advapi32 -lgfortrangcc.exe: error: unrecognized command line option '--exclude-libs=libpthread.a'make[1]: *** [Makefile:94: ../libopenblas.dll] Error 1make[1]: Leaving directory '/home/david/xianyi-OpenBLAS-3f6398a/exports'make: *** [Makefile:102: shared] Error 2

From: qwer1304@hotmail.com
To: reply@reply.github.com
Subject: RE: [xianyi/OpenBLAS] Incorrect handling of linker option (#920)
Date: Sat, 9 Jul 2016 11:56:38 +0300




This is the log with the error from a fresh install

Date: Fri, 8 Jul 2016 17:25:22 -0700
From: notifications@github.com
To: OpenBLAS@noreply.github.com
CC: qwer1304@hotmail.com; author@noreply.github.com
Subject: Re: [xianyi/OpenBLAS] Incorrect handling of linker option (#920)

unzip latast release in a new directory ;)


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

--- a/c_check
+++ b/c_check
@@ -234,6 +234,11 @@ $linker_a = "";
 	    $linker_L .= "-Wl,". $flags . " "
 	    }
 
+	if ($flags =~ /^\--exclude-libs/) {
+	    $linker_L .= "-Wl,". $flags . " ";
+	    $flags="";
+	   }
+
 	if (
 	    ($flags =~ /^\-l/)
 	    && ($flags !~ /gfortranbegin/)

--- a/f_check
+++ b/f_check
@@ -283,6 +283,12 @@ if ($link ne "") {
 	    $linker_L .= "-Wl,". $flags . " ";
         }
 
+	if ($flags =~ /^\--exclude-libs/) {
+	    $linker_L .= "-Wl,". $flags . " ";
+	    $flags="";
+	}
+
+
 	if ($flags =~ /^\-rpath\@/) {
 	    $flags =~ s/\@/\,/g;
 	    if ($vendor eq "PGI") {

